### PR TITLE
Linux

### PR DESCRIPTION
**Pull request by Debajyati**
## Roadmap URL
https://roadmap.sh/linux
## Nodes
vim
Bootloaders
Mounts
Shell Programming and Debugging in Bash

- Improved the content of the nodes - `shell programming` ,`vim` and `bootloaders`. 
- Added relevant resource links to the nodes - `mounts`, `bootloaders`, `shell programming`, `debugging`, `vim`

### CODE REVIEW

**Review by dansholds — CHANGES_REQUESTED** [review]
lets keep this just to adding links, you're adding more copy to old content that already doesn't abide by the newer contribution guidelines.

**Comment by Debajyati** [comment]
Okay. Just as you command. Sir! 🫡

**Comment by Debajyati** [comment]
Kindly review the changes again!

**Comment by offensive-vk** [comment]
So many youtube links and playlists in `114-shell-programming/index.md`?, topics links are too similar and content should be only in one para where possible.

**Comment by Debajyati** [comment]
> So many youtube links and playlists in `114-shell-programming/index.md`?, topics links are too similar and content should be only in one para where possible.

@offensive-vk done!

**Comment by offensive-vk** [comment]
Lastly, a proper pr title would be beneficial too.

File: src/data/roadmaps/linux/content/101-editing-files/100-vim.md [hunk 1]
```diff
@@ -23,4 +23,6 @@ Check out this [Github repo](https://github.com/iggredible/Learn-Vim?tab=readme-
 - [@article@Learn Vim Progressively](https://yannesposito.com/Scratch/en/blog/Learn-Vim-Progressively/)
 - [@video@Vim basics](https://www.youtube.com/watch?v=wACD8WEnImo&list=PLT98CRl2KxKHy4A5N70jMRYAROzzC2a6x&ab_channel=LearnLinuxTV)
 - [@article@Platform to practice Vim](https://vim-adventures.com/)
-- [@article@Vi Cheat Sheet](https://ryanstutorials.net/linuxtutorial/cheatsheetvi.php)
\ No newline at end of file
+- [@article@Vi Cheat Sheet](https://ryanstutorials.net/linuxtutorial/cheatsheetvi.php)
+- [@article@Vim Cheat Sheet](https://vim.rtorr.com/)
+- [@article@Learn Vimscript The Hard Way](https://learnvimscriptthehardway.stevelosh.com/)
```
Comment (dansholds) [inline src/data/roadmaps/linux/content/101-editing-files/100-vim.md:28]:
```suggestion
- [@article@Vim Cheat Sheet](https://vim.rtorr.com/)
- [@course@Learn Vimscript The Hard Way](https://learnvimscriptthehardway.stevelosh.com/)
```

File: src/data/roadmaps/linux/content/114-shell-programming/100-debugging.md [hunk 2]
```diff
@@ -16,4 +16,9 @@ Or, you can run a script in debug mode directly from the command line.
...
 ```
 
-These debugging tools and options can drastically help you in making your scripts more error-proof and efficient.
\ No newline at end of file
+These debugging tools and options can drastically help you in making your scripts more error-proof and efficient.
+
+Visit the following sources to learn more: 
+- [@official@Official Bashdb Documentation](https://bashdb.readthedocs.io/en/latest/)
+- [@article@How To Debug a Bash Shell Script Under Linux or UNIX](https://www.cyberciti.biz/tips/debugging-shell-script.html)
+- [@article@Debugging a Bash Script](https://www.baeldung.com/linux/debug-bash-script)
```
Comment (dansholds) [inline src/data/roadmaps/linux/content/114-shell-programming/100-debugging.md:24]:
```suggestion
```

File: src/data/roadmaps/linux/content/114-shell-programming/index.md [hunk 3]
```diff
@@ -14,5 +14,7 @@ echo "Hello, World!"
 The 'echo' command prints its argument, in this case "Hello, World!", to the terminal.
 
 Visit the following resources to learn more:
-
-- [@article@Bash Scripting Tutorial](https://ryanstutorials.net/bash-scripting-tutorial/)
\ No newline at end of file
+- [@article@Bash Scripting Tutorial](https://ryanstutorials.net/bash-scripting-tutorial/)
+- [@feed@Explore Top Posts About Bash](https://app.daily.dev/tags/bash)
+- [@video@Bash Scripting on Linux - YT Playlist](https://youtube.com/playlist?list=PLT98CRl2KxKGj-VKtApD8-zCqSaN2mD4w&si=MSehStqnhSqgoMSj)
+- [@article@9 Evil Bash Commands Explained](https://dev.to/devmount/9-evil-bash-commands-explained-4k5e)
```
Comment (dansholds) [inline src/data/roadmaps/linux/content/114-shell-programming/index.md:20]:
```suggestion

- [@video@Bash Scripting on Linux - YT Playlist](https://youtube.com/playlist?list=PLT98CRl2KxKGj-VKtApD8-zCqSaN2mD4w&si=MSehStqnhSqgoMSj)
```

**Review by dansholds — APPROVED** [review]

**Pull request merged**